core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala

```diff
@@ -91,6 +91,12 @@ private[spark] class StreamingPythonRunner(
       new BufferedInputStream(pythonWorker.get.channel.socket().getInputStream, bufferSize))
 
     val resFromPython = dataIn.readInt()
+    if (resFromPython != 0) {
+      val errMessage = PythonWorkerUtils.readUTF(dataIn)
+      throw new PythonException(s"Streaming Runner initialization failed" +
+        s" (returned $resFromPython). " +
+        s"Error message: $errMessage", null)
+    }
```

Comment:
can you add a proper error class for this? Or does the Python exception already have one?

Reply:
Created a new error class, PTAL